### PR TITLE
Aldaine

### PR DESCRIPTION
**Pull request by aldaineclarke**
Worked on the accounts endpoint...removed all schema middlewares and created methods in the account controller. 

user:
Pikachu
pikapika

should be able to create accounts and once accounts are deleted balance data is removed as well, should be able to find account by account number or account id
should be able to view all user accounts using username

### CODE REVIEW

File: controllers/account.controller.js [hunk 1]
```diff
@@ -1,53 +1,127 @@
...
             if(!ObjectId.isValid(id)) throw new Error ("ID does not match any accounts in database")
             if(Object.keys(data).length == 0){
                 return JSON.succes(res, "No data passed to update, file not updated",{}, 200);
             }
+            if(data.account_balance){
+                data.account_balance = undefined;
+            }
             let account = await Account.findByIdAndUpdate(id, data, {new:true});
+            if(!account) throw new Error("Account was not found with that ID");
+            account.account_bala;nce = undefined;
```
Comment (Dawnz) [inline controllers/account.controller.js]: typo. this prevented update function from working.

File: controllers/account.controller.js [hunk 2]
```diff
@@ -1,53 +1,127 @@
...
+            JSONResponse.error(res,"Error retrieving all accounts", error, 404);
         }
     }
 
+    // static populateAutoProperties = async (data)=>{
+    //     let wBalance = await this.generateBalance(data);
+    //     wBalance.account_number = await this.generateAccountNumber(wBalance.account_balance, 8);
+    //     return wBalance;
+    // }
     static updateAccount = async(req, res, next)=>{
```
Comment (Dawnz) [inline controllers/account.controller.js:96]: a user can edit the Enum values for account_type.

File: controllers/account.controller.js [hunk 3]
```diff
@@ -1,53 +1,127 @@
...
     static createAccount = async (req, res, next)=>{
+        let accountBal = "";
         try{
             let data = req.body;
+            data.id_type = data.id_type.toUpperCase();
             if(Object.keys(data).length == 0)throw new Error("No data was passed to create account");
-            const account = await new Account.create(data);
-            JSONResponse.success(res,"Account was successfully created", account, 201);
+            let account = new Account(data);
+            accountBal = await this.generateBalanceId(data);
```
Comment (Dawnz) [inline controllers/account.controller.js:17]: when i create an account the account balance is the accountId

Comment (Dawnz) [reply controllers/account.controller.js:17]: when this is fixed i can test the balance in transactions

File: SCHEMA/transaction.schema.js [hunk 4]
```diff
@@ -11,9 +13,17 @@ const transactionSchema = new Schema({
...
+    fname:{
+        type: String,
+        required:[true, "A first name must be provided"]
+    },
+    lname:{
         type: String,
-        required:[true, "A name must be provided"]
+        required:[true, "A last name must be provided"]
     },
     transaction_status:{
```
Comment (Dawnz) [inline SCHEMA/transaction.schema.js:28]: how will we use this. when will "pending" status change

**Review by Dawnz — CHANGES_REQUESTED** [review]
Plese see review comments

File: controllers/account.controller.js (hunk 3 above)
Comment (aldaineclarke) [reply controllers/account.controller.js:17]: it is the account balance id...they are linked to the account balance in the table..if you want me to populate it then thats fine too.

File: controllers/account.controller.js (hunk 2 above)
Comment (aldaineclarke) [reply controllers/account.controller.js:96]: I have set it to be required. Not sure what you mean by editing enum values though

File: controllers/account.controller.js (hunk 1 above)
Comment (aldaineclarke) [reply controllers/account.controller.js]: fixed..working on other reviews.

File: SCHEMA/transaction.schema.js (hunk 4 above)
Comment (aldaineclarke) [reply SCHEMA/transaction.schema.js:28]: Not sure at the moment..just having it here for future use

File: controllers/account.controller.js (hunk 3 above)
Comment (aldaineclarke) [reply controllers/account.controller.js:17]: projected the account balance and _id to be returned and assigned to the account balance

**Review by Dawnz — COMMENTED** [review]
the balance issue isnt resolved.
im not able to delete items with the id as account_balance because its greater than 100.
remove the ones created earlier with Id.

**Comment by aldaineclarke** [comment]
The deleting thing for the account was intentionally set. wouldn't want user to delete an account if it contains funds
aditionally you shouldn't be getting an account id for the balance...you would get an id yes, which represents the balance in the balance schema as well as the actual balance

**Comment by Dawnz** [comment]
if you are going to show the id then expose an endpoint that the balance can be shown by balanceId
Im only seeing transactions for the user you created. i should be able to delete the others as they have no money on the account
fix the merge conflicts and ill merge
@rdwayne876  @Geo-the-Pitbull  you can work on the  accounts, transaction and balance enpoint documentation when its merged.

**Pull request merged**